src/App.js

```diff
@@ -1,33 +1,61 @@
 // Packages
-import React from 'react';
+import { useState, useEffect } from 'react';
 import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
 
+// Firestore
+import { db } from './lib/firebase.js';
+
 // External Files
 import './App.css';
 
 // View Components
-import List from './pages/List/List';
-import AddItem from './pages/AddItem/AddItem';
+import ListView from './pages/ListView/ListView';
+import AddItemView from './pages/AddItemView/AddItemView';
 import NotFound from './pages/NotFound/NotFound';
 
-// Components
-import NavMenu from './components/NavMenu/NavMenu';
-
 function App() {
+  const [listId, setListId] = useState(null);
+
+  // on component mounting, look for token in local storage and use it to retrieve the list id
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+
+    if (token) {
+      // find the list in Firestore associated with the stored token from local storage
+      db.collection('lists')
+        .where('token', '==', token)
+        .get()
+        .then((querySnapshot) => {
+          // for some reason, extracting this array seems necessary to avoid errors?
+          const documents = querySnapshot.docs;
+
+          // check if the expected data is returned
+          if (
+            Array.isArray(documents) &&
+            documents.length &&
+            typeof documents[0] === 'object'
+          ) {
+            setListId(documents[0].id); // save the list id for later
+          }
+        })
+        .catch((error) => {
+          console.log('Error getting list: ', error);
+        });
+    }
+  }, []);
+
   return (
     <Router>
       <div className="App container">
         <Switch>
           <Route exact path="/">
-            <List />
+            <ListView listId={listId} />
```

Comment:
Not sure if we mentioned this anywhere, but we tacked "View" onto the names of the components in /pages because it was getting confusing with the new components we were adding (as I'm sure you can imagine)!

Reply:
These are really great ways to check. Sometimes libraries have a method for you to use so then your code would be easier to read. This is one of those use cases. Not a necessary change. Based on the [querySnapshot documentations](https://firebase.google.com/docs/reference/node/firebase.firestore.QuerySnapshot) @nick-zanetti  shared, you can use:
```javascript
if(!querySnaphort.empty) {
// save the list ide for later
}
```